lib/chef/provisioning/fog_driver/driver.rb

```diff
@@ -629,7 +633,14 @@ def create_ssh_transport(machine_spec, machine_options, server)
       end
 
       remote_host = nil
-      if machine_spec.location['use_private_ip_for_ssh']
+      if machine_options.has_key?(:bootstrap_network)
```

Comment:
This spot will need machine_spec.location['bootstrap_network'] when location is fixed.